packages/vite/src/node/utils.ts

```diff
@@ -622,6 +623,20 @@ export function copyDir(srcDir: string, destDir: string): void {
   }
 }
 
+export async function recursiveReaddir(dir: string): Promise<string[]> {
+  if (!fs.existsSync(dir)) {
+    return []
+  }
+  const dirents = await fsp.readdir(dir, { withFileTypes: true })
+  const files = await Promise.all(
+    dirents.map((dirent) => {
+      const res = path.resolve(dir, dirent.name)
+      return dirent.isDirectory() ? recursiveReaddir(res) : normalizePath(res)
+    }),
+  )
+  return Array.prototype.concat(...files)
```

Comment:
```suggestion
  return files.flat(1)
```
I think this describes the intention more precisely.